LX/src/main/java/heronarts/lx/LXSnapshotController.java

```diff
@@ -0,0 +1,161 @@
+package heronarts.lx;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.osc.OscMessage;
+
+public class LXSnapshotController extends LXComponent {
+
+  private Integer nextShapshotID = 1;
+
+  private final List<LXSnapshot> snapshots = new ArrayList<LXSnapshot>();
+  private final TreeMap<Integer, LXSnapshot> snapshotsDict = new TreeMap<Integer, LXSnapshot>();
```

Comment:
Should probably just be the array with reordering support, similar to how modulators are handled in the LXModulatorComponent.

This way you can also use addArray() for the component saving.

See:
https://github.com/heronarts/LX/blob/master/LX/src/main/java/heronarts/lx/LXModulatorComponent.java#L48

Scanning the array for lookup-by-index is fine, that's super-cheap and no risk of data inconsistency maintaining two containers of the same objects.

Reply:
Sounds good.